### PR TITLE
Only calculate Peff for a specific collider if [collider] is known

### PR DESCRIPTION
**Pull request by alongd**
### Motivation or Problem
When a library is loaded with a reaction that has a specific collider, e.g., `H + O2 (+H2O) <=> HO2 (+H2O)`, but the collider is not present in the system (e.g., in pyrolysis H2O is not present in the `y` concentration vector), RMG raises an error that the `y` vector cannot be indexed. 

### Description of Changes
Here we first check whether the index of the collider species appears in `y` before accessing it for the `Peff` determination, if not we ascribe `kf` of 0 (and hence also `kr`)

### Testing
Added a test with such case where a specific collider is not a species in the model.

### CODE REVIEW

**Comment by alongd** [comment]
@josemvdh, I added a test, please take a look

File: rmgpy/solver/simple.pyx [hunk 1]
```diff
@@ -397,11 +397,14 @@ cdef class SimpleReactor(ReactionSystem):
...
-                # Calculate effective pressure
-                Peff = P * y[self.species_index[specific_collider_species[i]]] / np.sum(y_core_species)
-                j = pdep_specific_collider_reaction_indices[i]
-                kf[j] = pdep_specific_collider_kinetics[i].get_rate_coefficient(T, Peff)
-                kr[j] = kf[j] / equilibrium_constants[j]
+                if len(y) > self.species_index[specific_collider_species[i]]:
+                    # Calculate the effective pressure
+                    Peff = P * y[self.species_index[specific_collider_species[i]]] / np.sum(y_core_species)
+                    j = pdep_specific_collider_reaction_indices[i]
+                    kf[j] = pdep_specific_collider_kinetics[i].get_rate_coefficient(T, Peff) or 0
```
Comment (josemvdh) [inline rmgpy/solver/simple.pyx:404]: I think the "or 0" can be removed now because of the else

**Comment by josemvdh** [comment]
I verified using a library with specific collider reactions that fails on master but RMG successfully terminates using this branch

**Comment by alongd** [comment]
@josemvdh, thanks for the comment, I made the requested modification and squashed the fix into the commit. Please let me know if you have any additional comments.

**Review by josemvdh — APPROVED** [review]

**Review by josemvdh — APPROVED** [review]

**Comment by alongd** [comment]
Thanks @josemvdh for serving as a reviewer!
@amarkpayne, would you mind taking a quick look and merging this PR? (or feel free to leave comments)

**Review by amarkpayne — DISMISSED** [review]

**Comment by amarkpayne** [comment]
@alongd Go ahead and rebase this PR and then feel free to merge it in, no further review needed

**Review by amarkpayne — APPROVED** [review]

**Pull request merged**